packages/liveblocks-core/src/room.ts

```diff
@@ -1011,7 +1015,33 @@ export function createRoom<
         throw new Error("Can't send message if socket is null");
       }
       if (context.socket.readyState === context.socket.OPEN) {
-        context.socket.send(JSON.stringify(messageOrMessages));
+        const message = JSON.stringify(messageOrMessages);
+        if (config.unstable_fallbackToHTTP) {
+          // if our message contains UTF-8, we can't simply use length. See: https://stackoverflow.com/questions/23318037/size-of-json-object-in-kbs-mbs
+          // if this turns out to be expensive, we could just guess with a lower value.
+          const size = new TextEncoder().encode(message).length;
+          if (
+            size > MAX_MESSAGE_SIZE &&
+            context.token?.raw &&
+            config.httpSendEndpoint
+          ) {
+            if (isTokenExpired(context.token.parsed)) {
+              return reconnect();
```

Comment:
@nvie if we hit this path, the messages here will be lost, I couldn't see a clean way to put the message back in the queue for processing after reconnect and I didn't see a way to await here. Any ideas on a clean way to do that?

Reply:
I don't think you have to worry about this case. All outgoing ops will be stored in an unacknowledged ops administration, which will not be cleared until the ops actually get acknowledged by the server. After a reconnect happens, the unacknowledged ops are resent automatically, in response to the `INITIAL_STORAGE_STATE` server message, see this part:

https://github.com/liveblocks/liveblocks/blob/cdb3cf41b6a5342b07df6f3b3a7520179d05ffae/packages/liveblocks-core/src/room.ts#L1692-L1697

So reconnecting here if the token is expired should already work, I think? Did it not?